assets/src/hooks/useChannel.ts

```diff
@@ -143,3 +143,82 @@ export const useCheckedChannel = <T, U>({
   ])
   return state
 }
+
+export const useCheckedTwoWayChannel = <T, U>({
+  socket,
+  topic,
+  event,
+  dataStruct,
+  parser,
+  loadingState,
+  initialMessage,
+}: {
+  socket: Socket | undefined
+  topic: string | null
+  event: string
+  initialMessage: any
+  dataStruct: Struct<T, any>
+  parser: (data: T) => U
+  loadingState: U
+}): [U, (event: string, payload: any) => void] => {
+  const [state, setState] = useState<U>(loadingState)
+  const [channel, setChannel] = useState<Channel | undefined>()
+
+  const onOk = ({ data: data }: { data: unknown }) => {
+    try {
+      assert(data, dataStruct)
+      setState(parser(data))
+    } catch (error) {
+      if (error instanceof StructError) {
+        Sentry.captureException(error)
+      }
+    }
+  }
+
+  const pushUpdate = useCallback(
+    (event: string, payload: any): void => {
+      if (channel) {
+        channel
+          .push(event, payload)
+          .receive("ok", (data: { data: unknown }) => {
+            onOk(data)
+          })
+      }
+    },
+    [channel]
+  )
+
+  useEffect(() => {
+    setState(loadingState)
+    let channel: Channel | undefined
```

Comment:
**suggestion:** I think I see why you want to have a separate local definition of `channel` so that you can run through the whole process of joining and attaching handlers before calling `setChannel`, but the variable shadowing with the `channel` associated with the state is a little confusing. Is it possible to come up with some other name? Or somehow rework it so that a different variable isn't required?